express-mock-af/services/reports.service.js

```diff
@@ -160,49 +141,32 @@ class ReportsService {
       .value();
   }
 
-  /**
-   * Filters reports based on the query parameters
-   *
-   * @param reportsList
-   * @param queryFilter
-   * @returns {*}
-   */
-  filterReports(reportsList, queryFilter) {
-    const clearQueryFilter = omitBy(queryFilter, isNil);
-    return reportsList.filter((report) => {
-      if (report.ReceptionReport) {
-        const flattenedReport = merge(
-          report.ReceptionReport,
-          report.ReceptionReport.QoeReport
-        );
-        return (
-          flattenedReport.recordingSessionId ===
-            clearQueryFilter.recordingSessionId &&
-          (flattenedReport.reportTime === clearQueryFilter.reportTime ||
-            clearQueryFilter.reportTime.includes(flattenedReport.reportTime))
-        );
-      } else if (report.consumptionReportingUnits) {
-        const filteredUnits = filter(
-          report.consumptionReportingUnits,
-          (unit) => {
-            const sameStartTime = queryFilter.startTime
-              ? unit.startTime === queryFilter.startTime
-              : true;
-            const sameDuration = queryFilter.duration
-              ? unit.duration === +queryFilter.duration
-              : true;
-            return sameStartTime && sameDuration;
-          }
-        );
-
-        const sameReportingClientId = queryFilter.reportingClientId
-          ? report.reportingClientId === queryFilter.reportingClientId
-          : true;
-        return sameReportingClientId && filteredUnits.length > 0;
-      }
-      return true;
-    });
-  }
+    /**
+     * Filters reports based on the query parameters
+     *
+     * @param reportsList
+     * @param queryFilter
+     * @returns {*}
+     */
+    filterReports(reportsList, queryFilter) {
+        const clearQueryFilter = omitBy(queryFilter, isNil);
+        return reportsList.filter(report => {
+            if(report.ReceptionReport) {
+                const flattenedReport = merge({}, report.ReceptionReport, report.ReceptionReport.QoeReport);
+                return isMatch(flattenedReport, clearQueryFilter);
```

Comment:
Das ist falsch und muss wieder zu 
```
return (flattenedReport.recordingSessionId === clearQueryFilter.recordingSessionId && (flattenedReport.reportTime === clearQueryFilter.reportTime || clearQueryFilter.reportTime.includes(flattenedReport.reportTime)));
```

geändert werden

Reply:
Dann macht die `merge`Methode aber keinen Sinn mehr. Warum genau sollte ich mergen, wenn ich danach alles einzeln validiere anstatt isMatch zu verwenden? 

Ich löse das in itself auf, in das isMatch können wir die includes logic nicht richtig integrieren.